transcom/mymove
# B21575-INT2-add order types

## Agility ticket
[B21575](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21575)

## Notes
This PR is a remake of prior PR.  Prior PR was: [14091](https://github.com/transcom/mymove/pull/14091)
Please see this PR for any prior comments or corrections.

## Summary
For the customer, 2 new order types were added:  Student Travel, Early Return of Dependents.  If either of these were selected, then "Has Dependents" would default to "yes" and be disabled so it can't be changed.  If another order type besides these 2 were selected, "Has Dependents" would be re-enabled and would be reset to nothing selected.

### How to test
In the .envrc.local file, add the following to turn on the feature flag:
export FEATURE_FLAG_ENABLE_ALASKA=true
restart server and client

Scenario 1: Creating a move
* login as a customer
* create a move
* at the move orders page ("tell us about your move orders")
  - set initial values all fields:
    - order type: Permanent Change of Station
    - remaining fields: any value is fine
* verification
  - verify: order type,  changing existing type to another existing type
    - change order type to local move
    - verify no fields are changed
  - verify: order type, changing existing type to new type (student travel)
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
  - verify: order type, changing new type (student travel) to existing type
    - change order type to permanent change of station
    - verify "are dependents included" defaults to no selection
    - verify "are dependents included" is enabled
  - verify: order type, changing existing type to new type (early return of dependents)
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
  - verify: order type, changing new type (early return of dependents) to existing type
    - change order type to permanent change of station
    - verify "are dependents included" defaults to no selection
    - verify "are dependents included" is enabled
  - verify: order type, changing new type (early return of dependents) to to new type (student travel)
    - change order type to early return of dependents
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
* finish creating the move w/a ppm shipment.

Scenario 2: edit existing orders
* login as a customer
* use the move just created in scenario 1
* click on "review your request"
* edit orders
* on the "edit orders" page, set the initial values the same as in scenario 1
* perform verification the same verification as in scenario 1.
* save the page

Scenario 3: turn off feature flag
In the .envrc.local file, add the following to turn on the feature flag:
export FEATURE_FLAG_ENABLE_ALASKA=false
restart server and client
* Create a move: On the orders page, verify neither Student Travel or Early Return of Dependents shows up in the drop down list
* Edit a move: On the orders page, verify neither Student Travel or Early Return of Dependents shows up in the drop down list

## ANDI Screenshots
Creating a move:
![image](https://github.com/user-attachments/assets/7eaf2b15-612b-4558-95d7-ecaeb938f7b8)

Editing existing orders:
![image](https://github.com/user-attachments/assets/b71fc586-4274-4705-87b6-80e0eea0ffc0)


## Testing Screenshots
N/A
